features/graph/dao/api/src/main/java/org/opennms/netmgt/topology/EdgeEntity.java

```diff
@@ -28,39 +28,46 @@
 
 package org.opennms.netmgt.topology;
 
-import javax.persistence.CascadeType;
+import java.util.Objects;
+
 import javax.persistence.DiscriminatorValue;
 import javax.persistence.Entity;
-import javax.persistence.JoinColumn;
-import javax.persistence.ManyToOne;
 
 @Entity
 @DiscriminatorValue("edge")
 public class EdgeEntity extends AbstractGraphEntity {
 
-    // TODO MVR we don't need this. This could be a property instead (due to vertex-ref), so source or target may not exist in the table
-    @ManyToOne(optional = false, cascade = CascadeType.ALL)
-    @JoinColumn(name="source_vertex_id", referencedColumnName = "id", nullable = false, updatable = true)
-    private VertexEntity source;
+    // TODO: Patrick: discuss with Markus if we want to have source and target as properties?
+    // Advantage: we can use the current implementation for storage.
+    // Disadvantage: we need to filter out the source/target properties when converting to GenericEdge since these are
+    // there object attributes not properties (and should be since an edge is defined by them)
```

Comment:
I am not sure about this.
Both variants have pros and cons.
The current graph structure is not very (postgres) database friendly. 
The only hard relation we enforce so far is: 
 - graph container -> graph
 - graph -> graph element
 - graph element -> property
 - edge -> source and target

Having source and target reference as a property is an option.
If we follow that route, we should make it part of the Converter and support it as a full type and not manually converting them.

If we use the more (postgres) database friendly approach, we should add a `VertexRefEntity` object and use that instead. 

I don't have a strong opinion, but leaning slightly towards property.
But let's have a call about this.